tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,741 @@
+
+RETURNS: 
+ - an array of hash references. Each hash has one key: C<minc_location> => location (absolute path)
+ of a MINC file found in table C<mri_protocol_violated_scans>.
+
+=cut
+sub getMriProtocolViolatedScansFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+    
+    # Get FileID and File path of each files in files directly tied
+    # to $tarchiveId
+    $query = 'SELECT minc_location FROM mri_protocol_violated_scans WHERE TarchiveID = ?';
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+    # Make sure all paths are absolute paths
+    foreach(@$filesRef) {
+        $_->{'minc_location'} = "$dataDirBasePath/trashbin/$_->{'minc_location'}" unless $_->{'minc_location'} =~ /^\//;
+    } 
+    
+    return $filesRef;   
+}
+
+=pod
+
+=head3 getMriViolationsLogFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Get the absolute paths of all the files associated to an archive that are listed in 
+table C<mri_violations_log>.
+
+INPUTS:
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting C<dataDirBasePath>.
+
+RETURNS: 
+ an array of hash references. Each hash has one key: C<MincFile> => location (absolute path)
+ of a MINC file found in table C<mri_violations_log>.
+
+=cut
+sub getMriViolationsLogFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+    
+    # Get file path of each files in files directly tied
+    # to $tarchiveId
+    $query = 'SELECT MincFile FROM mri_violations_log WHERE TarchiveID = ?';
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+    # Make sure all paths are absolute paths
+    foreach(@$filesRef) {
+        $_->{'MincFile'} = "$dataDirBasePath/trashbin/$_->{'MincFile'}" unless $_->{'MincFile'} =~ /^\//;
+    } 
+    
+    return $filesRef;   
+}
+
+=pod
+
+=head3 getMRICandidateErrorsFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Get the absolute paths of all the files associated to an archive that are listed in 
+table C<MRICandidateErrors>.
+
+INPUTS:
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting C<dataDirBasePath>.
+
+RETURNS: 
+ - an array of hash references. Each hash has one key: C<MincFile> => location (absolute path)
+ of a MINC file found in table C<MRICandidateErrors>.
+
+=cut
+sub getMRICandidateErrorsFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+    
+    # Get file path of each files in files directly tied
+    # to $tarchiveId
+    $query = 'SELECT MincFile FROM MRICandidateErrors WHERE TarchiveID = ?';
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+    # Make sure all paths are absolute paths
+    foreach(@$filesRef) {
+        $_->{'MincFile'} = "$dataDirBasePath/trashbin/$_->{'MincFile'}" unless $_->{'MincFile'} =~ /^\//;
```

Comment:
with my PR that fixes the violation tables, the path stored in the 3 violation tables are relative to `data_dir` so it needs to be removed from the path. I can't remove the `trashbin` in my PR as sometimes, in the `mri_violations_log` table, the path stored are the path in the `assembly` folder and it is relative to `data_dir` so I think we should update your PR instead.

Reply:
Ok, I'll assume all path in tables `mri_violations_log`, `mri_protocol_violated_scans` and `MRICandidateErrors` are relative to `data_dir` (not `data_dir/trashbin`)